src/Language/PureScript/Bridge/Printer.hs

```diff
@@ -71,10 +71,24 @@ importLineToText l = "import " <> importModule l <> " (" <> typeList <> ")"
     typeList = T.intercalate ", " (Set.toList (importTypes l))
 
 sumTypeToText :: SumType 'PureScript -> Text
-sumTypeToText (SumType t cs) = T.unlines $
+sumTypeToText st@(SumType t cs) = T.unlines $
     "data " <> typeInfoToText True t <> " ="
   : "    " <> T.intercalate "\n  | " (map (constructorToText 4) cs)
-  : [ "\nderive instance generic" <> _typeName t <> " :: Generic " <> _typeName t ]
+  : [ "\nderive instance generic" <> _typeName t <> " :: " <> genericConstrains <> genericInstance t ]
+  where
+    genericInstance = ("Generic " <>) . typeInfoToText False
+    genericConstrains
+        | stpLength == 0 = mempty
+        | otherwise = (<> " => ") $
+            if stpLength == 1
+                then genericConstrainsInner
+                else bracketWrap genericConstrainsInner
+    genericConstrainsInner = T.intercalate ", " $ map genericInstance sumTypeParameters
+    stpLength = length sumTypeParameters
+    bracketWrap x = "(" <> x <> ")"
+    sumTypeParameters = filter isTypeParameter $ getUsedTypes st
+    -- TODO: I find this stupid. Would much rather like some other way to make this check?
+    isTypeParameter tp = T.toLower (_typeName tp) == _typeName tp
```

Comment:
Ah, I see. Will update this in a min
Dana 6. 9. 2016. 10:43 "Robert Klotzner" <notifications@github.com> je
napisao/la:
>
> In src/Language/PureScript/Bridge/Printer.hs:
>
> > -  : [ "\nderive instance generic" <> _typeName t <> " :: Generic " <>
_typeName t ]
> > +  : [ "\nderive instance generic" <> _typeName t <> " :: " <>
genericConstrains <> genericInstance t ]
> > +  where
> > +    genericInstance = ("Generic " <>) . typeInfoToText False
> > +    genericConstrains
> > +        | stpLength == 0 = mempty
> > +        | otherwise = (<> " => ") $
> > +            if stpLength == 1
> > +                then genericConstrainsInner
> > +                else bracketWrap genericConstrainsInner
> > +    genericConstrainsInner = T.intercalate ", " $ map genericInstance
sumTypeParameters
> > +    stpLength = length sumTypeParameters
> > +    bracketWrap x = "(" <> x <> ")"
> > +    sumTypeParameters = filter isTypeParameter $ getUsedTypes st
> > +    -- TODO: I find this stupid. Would much rather like some other way
to make this check?
> > +    isTypeParameter tp = T.toLower (_typeName tp) == _typeName tp
>
> Just filter used types by whether they are contained in the sumtype's
TypeInfo's _typeParameters.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.